### PR TITLE
[2/n] add newtype wrappers to ensure config identifier validity

### PR DESCRIPTION
**Pull request by sunshowers**
I'm going to add some more identifiers soon, and I wanted to make sure at
deserialize time that they were somewhat well-formed. We restrict ourselves to
ASCII at the moment, which is fine.

Add three newtype wrappers -- one each for package, service and preset names.

Depends on:

* #70

### CODE REVIEW

File: src/config/identifier.rs [hunk 1]
```diff
@@ -0,0 +1,168 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::{borrow::Cow, fmt, str::FromStr};
+
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+/// A unique identifier for a configuration parameter.
```
Comment (smklein) [inline src/config/identifier.rs:10]: Makes sense to codify this a little better, thanks!

File: src/config/identifier.rs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+impl FromStr for ConfigIdent {
+    type Err = InvalidConfigIdent;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::new(s)
+    }
+}
+
+impl<'de> Deserialize<'de> for ConfigIdent {
```
Comment (smklein) [inline src/config/identifier.rs:90]: Why manually implement Deserialize but not Serialize? 

Is it worth adding a test that we can serialize -> deserialize -> re-serialize?

File: src/config/imp.rs [hunk 3]
```diff
@@ -12,10 +12,12 @@ use std::path::Path;
 use thiserror::Error;
 use topological_sort::TopologicalSort;
 
+use super::ConfigIdent;
+
 /// Describes a set of packages to act upon.
 ///
 /// This structure maps "package name" to "package"
```
Comment (smklein) [inline src/config/imp.rs:19]: Do you think it's worth it to make a thin wrapper around `ConfigIdent ` to label this as a `PackageName` specifically?

I'm assuming that with other configuration parameter usage, there will be other uses of `ConfigIdent` that are semantically different.

(If you want to punt this beyond this PR, I'm also fine with that)

**Review by smklein — APPROVED** [review]

File: src/config/identifier.rs (hunk 2 above)
Comment (sunshowers) [reply src/config/identifier.rs:90]: `Deserialize` must go through validation, while serialization doesn't have to. Definitely worth writing a test for.

Comment (sunshowers) [reply src/config/identifier.rs:177]: Added tests and a comment explaining this.

File: src/config/imp.rs (hunk 3 above)
Comment (sunshowers) [reply src/config/imp.rs:19]: Per discussion in DMs, ended up creating three newtypes -- one each for package names, service names and preset names.

**Pull request merged**